src/Microsoft.Diagnostics.DebugServices.Implementation/CommandService.cs

```diff
@@ -242,36 +242,30 @@ public void AddCommands(Type type, Func<IServiceProvider, object> factory)
         {
             if (type.IsClass)
             {
-                for (Type baseType = type; baseType != null; baseType = baseType.BaseType)
+                factory ??= (services) => Utilities.CreateInstance(type, services);
+
+                // Only look at the actual type and not any the base types for command attributes
```

Comment:
Is there a particular reason for no longer looking at any of the basetypes? Was it just unnecessary?

Reply:
Not necessary and causes problems if multiple commands (marked with the CommandAttribute) derive from a base class that also has a CommandAttribute.  The base class has already been or will be added so we don't want to do it again. We don't have any examples right now but I ran into a case playing around with some new commands in different branch.